src/Raven.Client/Documents/Commands/Batches/DeleteCommandData.cs

```diff
@@ -9,24 +9,38 @@ namespace Raven.Client.Documents.Commands.Batches
     public class DeleteCommandData : ICommandData
     {
         public DeleteCommandData(string id, string changeVector)
+            :this(id, changeVector, changeVector)
+        {
+            
+        }
+        public DeleteCommandData(string id, string changeVector, string oldChangeVector)
         {
             Id = id ?? throw new ArgumentNullException(nameof(id));
             ChangeVector = changeVector;
+            OldChangeVector = oldChangeVector;
         }
 
+        public string OldChangeVector { get; }
```

Comment:
If I understand correctly, now we will always send the change vector for concurrency check against the atomic guard. 
Maybe instead, we should add `ConcurrencyCheckMode` enum which is more explicit and clear

Reply:
That would break older clients, however, we need to send a different value, which won't cause issues.

Reply:
It might break when we use new client but old server. 
But the client already knows the server version `LastServerVersion`, so we can decided whether to send the change vector or not.

Reply:
Pretty sure that we should ignore a property that we don't know, no?
That said, a 5.2 client isn't supported for 5.0, but we'll need to check that.

Reply:
Karmel is talking about:
a) adding new property called ConcurrencyCheckMode and using that in 5.2 server along with ChangeVector instead of having OldChangeVector
b) when communicating with 5.1 or earlier server detect that and use the ChangeVector property appropriately

Reply:
I think so as well.. 
So 5.2 with 5.2 is easy, since we will always send the `ConcurrencyCheckMode` and know what to do with that..
For 5.2 server talking with 5.0 client not sending an the `ConcurrencyCheckMode` will mean Optimistic check by default.
For 5.0 and 5.2 client, the client can recognize the server version and not send the change vector if not needed.

If feasible, I think it is a much cleaner solution

Reply:
Okay, sound reasonable.